src/meta/src/hummock/compaction/overlap_strategy.rs

```diff
@@ -157,13 +155,47 @@ fn check_table_overlap(key_range: &KeyRange, table: &SstableInfo) -> bool {
     key_range.full_key_overlap(&other)
 }
 
+fn check_key_vnode_overlap(info: &SstableInfo, table: &SstableInfo) -> bool {
```

Comment:
Please add some explanation for it

Reply:
And unit tests...

Reply:
> And unit tests... May use the example in [design doc](https://singularity-data.quip.com/WpJXA1O4B82s#WeQABA76Lcn) as an unit test.

recursion not implemented

Reply:
> recursion not implemented

You mean we can only choose lower layer ssts according to upper layer, but the opposite is not supported?

Reply:
> > recursion not implemented
> 
> You mean we can only choose lower layer ssts according to upper layer, but the opposite is not supported?

If so, that would be a problem of correctness...

Reply:
... but won't be triggered without scale out/in 🤣

Reply:
> > > recursion not implemented
> > 
> > 
> > You mean we can only choose lower layer ssts according to upper layer, but the opposite is not supported?
> 
> If so, that would be a problem of correctness...

currently no tombstone issue now